tensorrt_llm/_torch/pyexecutor/kv_cache_manager_v2.py

```
@@ -841,6 +841,27 @@ def append_to_kv_heads_per_layer(
+            # without dividing the per-node budget by the local rank count,
+            # N co-located ranks each reserving a device-quota-sized block
+            # can OOM the host (observed on GB300 NVL72 with 4 ranks/node
+            # and ~170GiB device quota each on a 975GiB node).
+            local_ranks = 1
+            try:
+                if mapping.world_size > 1:
+                    hostnames = Distributed.get(mapping).allgather(socket.gethostname())
+                    if hostnames:
+                        self_hostname = socket.gethostname()
+                        local_ranks = max(1, sum(1 for h in hostnames if h == self_hostname))
+            except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)